vivisect/qt/main.py

```diff
@@ -435,28 +537,73 @@ def _menuToolsPython(self):
         self.vqBuildDockWidget('VQPythonView', area=QtCore.Qt.RightDockWidgetArea)
 
     def _menuViewStrings(self):
-        self.vqBuildDockWidget('VQVivStringsView', area=QtCore.Qt.RightDockWidgetArea)
-
+        self.newStringsView()
     def _menuViewStructs(self):
-        self.vqBuildDockWidget('VQVivStructsView', area=QtCore.Qt.RightDockWidgetArea)
-
+        self.newStructsView()
     def _menuViewSegments(self):
-        self.vqBuildDockWidget('VQVivSegmentsView', area=QtCore.Qt.RightDockWidgetArea)
-
+        self.newSegmentsView()
     def _menuViewImports(self):
-        self.vqBuildDockWidget('VQVivImportsView', area=QtCore.Qt.RightDockWidgetArea)
-
+        self.newImportsView()
     def _menuViewExports(self):
-        self.vqBuildDockWidget('VQVivExportsView', area=QtCore.Qt.RightDockWidgetArea)
-
+        self.newExportsView()
     def _menuViewFunctions(self):
-        self.vqBuildDockWidget('VQVivFunctionsView', area=QtCore.Qt.RightDockWidgetArea)
-
+        self.newFunctionsView()
     def _menuViewNames(self):
-        self.vqBuildDockWidget('VQVivNamesView', area=QtCore.Qt.RightDockWidgetArea)
-
+        self.newNamesView()
     def _menuViewMemory(self):
-        self.vqBuildDockWidget('VQVivMemoryView', area=QtCore.Qt.TopDockWidgetArea)
+        self.newMemoryView()
+    def _menuViewFuncGraph(self):
+        self.newFuncGraphView()
+    def _menuViewSymboliks(self):
+        self.newSymbolikFuncView()
+
+    @idlethread
```

Comment:
yes, i've run this code a lot.
idlethread is *required* since it makes changes to the GUI.
if we make these calls from other tools running outside the GUI thread, baaaaad things happen.  ie. core-dump.

Reply:
:+1: